### ISSUE TITLE
Photon Controller deployment doesn´t work with a free Hypervisor license

### ISSUE THREAD

**Issue by gregorkistler** [vmware/photon-controller#31]
Hi,

topic says it all. I´ve staged a few ESXi hosts within the last days and wondered why the deployment of the Photon Controller was only possible on one host and not on few additional ones, where it always failed at the second stage called 'Prepare deployment'. 

Logs of esxcloud/deployer shows plenty of re-occuring errors like

`ERROR [2016-05-26 09:46:59,979] com.vmware.photon.controller.common.thrift.WrappedCallback: Releasing client on error
! java.net.ConnectException: Connection refused
! at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[na:1.8.0_51-BLFS]
! at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717) ~[na:1.8.0_51-BLFS]
! at org.apache.thrift.transport.TNonblockingSocket.finishConnect(TNonblockingSocket.java:208) ~[libthrift-0.9.3.jar:0.9.3]
! at org.apache.thrift.async.TAsyncMethodCall.doConnecting(TAsyncMethodCall.java:267) ~[libthrift-0.9.3.jar:0.9.3]
! at org.apache.thrift.async.TAsyncMethodCall.transition(TAsyncMethodCall.java:183) ~[libthrift-0.9.3.jar:0.9.3]
! at org.apache.thrift.async.TAsyncClientManager$SelectThread.transitionMethods(TAsyncClientManager.java:143) [libthrift-0.9.3.jar:0.9.3]
! at org.apache.thrift.async.TAsyncClientManager$SelectThread.run(TAsyncClientManager.java:113) [libthrift-0.9.3.jar:0.9.3]`

followed by a final error

`ERROR [2016-05-26 09:47:12,197] com.vmware.photon.controller.deployer.service.client.DeploymentWorkFlowServiceClient: Deployment task failed: {"managementVmImageFile":"/var/photon/images/photon-management-vm-disk1.vmdk","taskState":{"stage":"FAILED","isDirect":false,"failure":{"message":"The agent on host <IP_ADDRESS> failed to become ready after installation after 60 retries","stackTrace":`

After I figured out that the only difference was the assigned license I´ve deleted the free Hypervisor license and reverted to the trial license and et voila .. I was able to deploy it.

Question: simply a bug or does it rely on a few services/features which are disabled while using the free Hypervisor license?

**Comment by mwest44** [comment]
The write APIs in the free version of esxi are disabled.   So it does not work with PC.  We are working on a solution for this and hope to have resolved this prior to GA

Sent from my iPhone

On May 26, 2016, at 4:40 AM, Gregor Kistler <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi,

topic says it all. I´ve staged a few ESXi hosts within the last days and wondered why the deployment of the Photon Controller was only possible on one host and not on few additional ones, where it always failed at the second stage called 'Prepare deployment'.

Logs of esxcloud/deployer shows plenty of re-occuring errors like

ERROR [2016-05-26 09:46:59,979] com.vmware.photon.controller.common.thrift.WrappedCallback: Releasing client on error
! java.net.ConnectException: Connection refused
! at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[na:1.8.0_51-BLFS]
! at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717) ~[na:1.8.0_51-BLFS]
! at org.apache.thrift.transport.TNonblockingSocket.finishConnect(TNonblockingSocket.java:208) ~[libthrift-0.9.3.jar:0.9.3]
! at org.apache.thrift.async.TAsyncMethodCall.doConnecting(TAsyncMethodCall.java:267) ~[libthrift-0.9.3.jar:0.9.3]
! at org.apache.thrift.async.TAsyncMethodCall.transition(TAsyncMethodCall.java:183) ~[libthrift-0.9.3.jar:0.9.3]
! at org.apache.thrift.async.TAsyncClientManager$SelectThread.transitionMethods(TAsyncClientManager.java:143) [libthrift-0.9.3.jar:0.9.3]
! at org.apache.thrift.async.TAsyncClientManager$SelectThread.run(TAsyncClientManager.java:113) [libthrift-0.9.3.jar:0.9.3]

followed by a final error

ERROR [2016-05-26 09:47:12,197] com.vmware.photon.controller.deployer.service.client.DeploymentWorkFlowServiceClient: Deployment task failed: {"managementVmImageFile":"/var/photon/images/photon-management-vm-disk1.vmdk","taskState":{"stage":"FAILED","isDirect":false,"failure":{"message":"The agent on host <IP_ADDRESS> failed to become ready after installation after 60 retries","stackTrace":

After I figured out that the only difference was the assigned license I´ve deleted the free Hypervisor license and reverted to the trial license and et voila .. I was able to deploy it.

Question: simply a bug or does it rely on a few services/features which are disabled while using the free Hypervisor license?

-
